### PR TITLE
Issue 150: Long Short-Term Memory (LSTM) Binary Classifier for texts

### LINKED ISSUE #150: Add primitive for Sequence classification with LSTM
Related to #121 

The architecture would be:

```
Embedding ---> LSTM --> Dropout --> Dense (sigmoid)
```

You can find an example of this [here](https://keras.io/getting-started/sequential-model-guide/#examples) in the **Sequence classification with LSTM** section:
```python
from keras.models import Sequential
from keras.layers import Dense, Dropout
from keras.layers import Embedding
from keras.layers import LSTM

max_features = 1024

model = Sequential()
model.add(Embedding(max_features, output_dim=256))
model.add(LSTM(128))
model.add(Dropout(0.5))
model.add(Dense(1, activation='sigmoid'))

model.compile(loss='binary_crossentropy',
              optimizer='rmsprop',
              metrics=['accuracy'])

model.fit(x_train, y_train, batch_size=16, epochs=10)
score = model.evaluate(x_test, y_test, batch_size=16)
```

### PR DESCRIPTION
**Pull request by Hector-hedb12**
Resolves #150

### CODE REVIEW

File: mlprimitives/jsons/keras.Sequential.LSTMBinaryTextClassifier.json [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        "output": [
+            {
+                "name": "y",
+                "type": "array"
+            }
+        ]
+    },
+    "hyperparameters": {
+        "fixed": {
+            "classification": {
```
[the remark this answers is not in the document]
Comment (Hector-hedb12) [reply mlprimitives/jsons/keras.Sequential.LSTMBinaryTextClassifier.json]: @csala according to the [doc](https://keras.io/models/model/), verbose should be an integer:

> verbose: Integer. 0, 1, or 2. Verbosity mode. 0 = silent, 1 = progress bar, 2 = one line per epoch.

So, a bool might not work, right? :thinking:

File: mlprimitives/jsons/keras.Sequential.LSTMBinaryTextClassifier.json [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            },
+            "dropout_rate": {
+                "type": "float",
+                "default": 0.5,
+                "range": [
+                    0.0,
+                    0.75
+                ]
+            },
+            "batch_size": {
```
[the remark this answers is not in the document]
Comment (Hector-hedb12) [reply mlprimitives/jsons/keras.Sequential.LSTMBinaryTextClassifier.json]: 928cfab

File: mlprimitives/jsons/keras.Sequential.LSTMBinaryTextClassifier.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hector-hedb12) [reply mlprimitives/jsons/keras.Sequential.LSTMBinaryTextClassifier.json]: 1b2b7f4

[the remark this answers is not in the document]
Comment (csala) [reply mlprimitives/jsons/keras.Sequential.LSTMBinaryTextClassifier.json]: Well, it actually does work. I guess `True/False` is internally interpreted as `1/0`. Feel free to change it to `int` type.

File: mlprimitives/jsons/keras.Sequential.LSTMBinaryTextClassifier.json [hunk 3]
```diff
@@ -67,6 +67,14 @@
                 "type": "int",
                 "default": 20
             },
+            "verbose": {
+                "type": "int",
+                "default": 1,
```
Comment (csala) [inline mlprimitives/jsons/keras.Sequential.LSTMBinaryTextClassifier.json]: Please set the default to 0

Comment (Hector-hedb12) [reply mlprimitives/jsons/keras.Sequential.LSTMBinaryTextClassifier.json]: 6a366fa

**Comment by csala** [comment]
@Hector-hedb12 The primitive looks good and can be merged, but I'm afraid the pipeline does not.

According to the Keras examples, this architecture is suitable to be used with tokenized sequences as input.
However, the pipeline that you included has two problems:
1. It uses a StringVectorizer, which produces a token count matrix instead of token sequences.
2. It is uses a validation dataset (personae) that contains more information a part from the texts, but this primitive is intended to receive as input nothing but the sequence tokens.

I think that here we need to:
1. Add a new dataset for binary text classification
2. Use the keras.Tokenizer instead of the StringVectorizer in the preprocessing steps.

In order to avoid having this PR stuck I will merge it as it is right now, but I will create a new issue to improve the pipeline.

**Pull request merged**